client/src/pages/Feed.js

```diff
@@ -406,15 +423,49 @@ const Feed = (props) => {
         });
       }).observe(node);
     },
-    [postsDispatch, isLoading, loadMore],
+    [postsDispatch, loadMore, isLoading],
   );
 
   useEffect(() => {
+    let observer;
     if (bottomBoundaryRef.current) {
-      scrollObserver(bottomBoundaryRef.current);
+      observer = scrollObserver(bottomBoundaryRef.current);
     }
+    return () => {
+      observer && observer.disconnect();
+    };
   }, [scrollObserver, bottomBoundaryRef]);
 
+  const postDelete = async (post) => {
+    let deleterResponse;
+    const endPoint = `/api/posts/${post._id}`;
+
+    if (
+      isAuthenticated &&
+      user &&
+      (user._id === post.author.id || user.id === post.author.id)
+    ) {
+      try {
+        deleterResponse = await axios.delete(endPoint);
+        if (deleterResponse && deleterResponse.data.success === true) {
+          const allPosts = {
+            ...postsList,
+          };
+          delete allPosts[post._id];
+
+          await postsDispatch({
+            type: SET_POSTS,
+            posts: allPosts,
+          });
+        }
+      } catch (error) {
+        console.log({
+          error,
```

Comment:
weird formatting from ESLINT probably? or is it this one the right way? I saw others above that are different than this one.

Reply:
It was the linter. Yeah, Robin mentioned the linter was doing weird things lately, idk if he figured out why.